packages/ra-data-simple-rest/src/index.spec.ts

```diff
@@ -68,6 +68,65 @@ describe('Data Simple REST Client', () => {
             expect(result.total).toEqual(42);
         });
     });
+    describe('getOne', () => {
+        it('should allow numeric id in path', async () => {
+            const httpClient = jest.fn().mockResolvedValue({ id: 'Post#123' });
```

Comment:
This may cause a warning as the requested `id` and the returned `id` aren't the same.

Reply:
Fixing.